misk/src/main/kotlin/misk/web/interceptors/RequestBodyLoggingInterceptor.kt

```diff
@@ -5,61 +5,46 @@ import misk.ApplicationInterceptor
 import misk.Chain
 import misk.MiskCaller
 import misk.logging.getLogger
-import misk.random.ThreadLocalRandom
 import misk.scope.ActionScoped
 import javax.inject.Inject
 import javax.inject.Singleton
-import kotlin.reflect.full.findAnnotation
 
 private val logger = getLogger<RequestBodyLoggingInterceptor>()
 
 /**
- * Logs request and response information for an action.
+ * Stores request and response information for an action in a ThreadLocal, to be logged
+ * in [RequestLoggingInterceptor]
+ *
  * Timing information doesn't count time writing the response to the remote client.
  */
-class RequestBodyLoggingInterceptor internal constructor(
+class RequestBodyLoggingInterceptor @Inject internal constructor(
   private val action: Action,
-  private val sampling: Double,
   private val caller: ActionScoped<MiskCaller?>,
-  private val random: ThreadLocalRandom
+  private val bodyCapture: RequestResponseCapture
 ) : ApplicationInterceptor {
 
   @Singleton
   class Factory @Inject internal constructor(
     private val caller: @JvmSuppressWildcards ActionScoped<MiskCaller?>,
-    private val random: ThreadLocalRandom
+    private val bodyCapture: RequestResponseCapture
   ) : ApplicationInterceptor.Factory {
     override fun create(action: Action): ApplicationInterceptor? {
-      val logRequestResponse = action.function.findAnnotation<LogRequestResponse>() ?: return null
-      require(0.0 < logRequestResponse.sampling && logRequestResponse.sampling <= 1.0) {
-        "${action.name} @LogRequestResponse sampling must be in the range (0.0, 1.0]"
-      }
-      if (!logRequestResponse.includeBody) {
-        return null
-      }
```

Comment:
I think the block above should be kept. If body sampling isn't enabled, then you shouldn't install this interceptor